src/io_uring.c

```diff
@@ -9,73 +9,78 @@
 
 #define IO_URING_DEPTH 256 /* io_uring instance queue depth. */
 
-static size_t io_uring_queue_len = 0;
+struct io_uring_context {
+    struct io_uring *ring;
```

Comment:
My suggestion was to embed the struct io_uring itself here, instead of a pointer. Just one allocation, not two. It better I guess, wdyt?

Use like this:

```C
struct io_uring_context *uring_context = zmalloc(sizeof(struct io_uring_context));
struct io_uring *ring = &(uring_context->ring);
...
io_uring_queue_init_params(IO_URING_DEPTH, ring, &params);
```

Reply:
Should we model the error handling model used by `connection` and embed an errno field in `io_uring_context`?

https://github.com/valkey-io/valkey/blob/unstable/src/connection.h#L122

Reply:
> Should we model the error handling model used by connection and embed an errno field in io_uring_context?

`struct connection` is transparent, our `struct io_uring_context` is opaque, do you want to publish the last_errno to client by a new function ?
```  c
int getLastErrno(io_uring_context *uring_instance) ;
```